src/_pytest/assertion/rewrite.py

```diff
@@ -128,12 +131,24 @@ def create_module(
         return None  # default behaviour is fine
 
     def exec_module(self, module: types.ModuleType) -> None:
-        assert module.__spec__ is not None
-        assert module.__spec__.origin is not None
-        fn = Path(module.__spec__.origin)
+        co = self.get_code(module.__name__)
+        exec(co, module.__dict__)
+
+    def get_code(self, fullname: str) -> Optional[types.CodeType]:
+        """
+        Returns the code for a rewritten test file.
+        """
+        # The type annotation is *Optional[...]* to match the declaration
+        # in importlib.abc.InspectLoader
```

Comment:
this comment is unnecessary -- the type checker will enforce this for us (remove the other one as well)